### PR TITLE
Introduce `ServerRequestCreator` and `UploadedFileCreator` class implementation with tests.

### PR DESCRIPTION
**Pull request by terabytesoftw**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | /❌

## Summary by CodeRabbit

* **New Features**
  * Added the ability to create HTTP server requests from PHP global variables, supporting both SAPI and worker environments.
  * Introduced support for handling single and multiple file uploads, including nested file structures, with robust validation and error handling.

* **Bug Fixes**
  * Enhanced error messages for file upload and response stream validation to provide clearer feedback on invalid input.

* **Tests**
  * Added comprehensive test coverage for server request and uploaded file creation, including edge cases and error scenarios.
  * Improved test infrastructure with better temporary file management and new factory helpers for easier test setup.

### CODE REVIEW

File: src/creator/UploadedFileCreator.php [hunk 1]
```diff
@@ -0,0 +1,495 @@
...
+        string|null $type = null,
+    ): UploadedFileInterface {
+        return $this->uploadedFileFactory->createUploadedFile(
+            $this->streamFactory->createStreamFromFile($tmpName),
+            $size,
+            $error,
+            $name,
+            $type,
+        );
+    }
```
[the remark this answers is not in the document]
Comment (terabytesoftw) [reply src/creator/UploadedFileCreator.php:337]: fixed.
